playlist.c

```diff
@@ -837,11 +882,14 @@ bool playlist_push(playlist_t *playlist,
       return false;
    }
 
-   /* Get 'real' path */
    if (!string_is_empty(entry->path))
    {
+      /* Get 'real' path */
       strlcpy(real_path, entry->path, sizeof(real_path));
       playlist_resolve_path(PLAYLIST_SAVE, real_path, sizeof(real_path));
+
+      /* Build relative path, if possible*/
+      path_build_relative_path(relative_path, entry->path, playlist->config.base_content_directory, PATH_MAX_LENGTH);
```

Comment:
No, let's not perform unnecessary function calls - wrap this in an `if (playlist->config.relative_paths)`

Reply:
Done.